packages/editor/src/extensions/code-block/code-block.ts

```diff
@@ -536,6 +537,15 @@ export const CodeBlock = Node.create<CodeBlockOptions>({
             l.filename === node.attrs.language ||
             l.alias?.some((a) => a === node.attrs.language)
         );
+
+        const cachedLanguage =
+          config.get<(typeof Languages)[number]>("codeBlockLanguage");
+        if (!languageDefinition && cachedLanguage && cachedLanguage.filename) {
+          this.editor.commands.updateAttributes(this.type, {
+            language: cachedLanguage.filename
+          });
+        }
```

Comment:
Thanks again for the thorough review!

I've added the language update from local storage logic here and it works. Please let me know if this is okay. Other place this exact logic works is in the `useEffect` inside the code block component.

Reply:
I just checked again and we can reuse `defaultLanguage` in the `Highlighter` to support this. It can be a function since the value can change.

However, the default language doesn't get stored in the code block attributes which you'll need to add support for in `appendTransaction`.

Reply:
Done, please review